src/Rings/mpoly-ideals.jl

```diff
@@ -822,7 +817,7 @@ end
 function primary_decomposition(
     I::MPolyIdeal{T}; 
     algorithm::Symbol=:GTZ, cache::Bool=true
-  ) where {U<:Union{AbsSimpleNumFieldElem, <:Hecke.RelSimpleNumFieldElem}, T<:MPolyRingElem{U}}
+  ) where {U<:Union{<:Hecke.RelSimpleNumFieldElem}, T<:MPolyRingElem{U}}
```

Comment:
```suggestion
  ) where {U<:Hecke.RelSimpleNumFieldElem, T<:MPolyRingElem{U}}
```

Reply:
Then you can add the as needed. But why is that an argument to "keep a spare `Union{...}` around? You wouldn't write `1+x` as `((1+x))` just in case you might later want to change it to `(2*(1+x))^3`, would you?